django/thunderstore/community/api/experimental/tests/test_api_package_listing.py

```diff
@@ -34,18 +34,27 @@ def test_api_experimental_package_listing_update_user_types(
         TestUserTypes.site_admin: True,
         TestUserTypes.superuser: True,
     }
+
+    error_message = {
+        TestUserTypes.no_user: ["Must be authenticated"],
+        TestUserTypes.unauthenticated: ["Must be authenticated"],
+        TestUserTypes.regular_user: ["User is missing necessary roles or permissions"],
+        TestUserTypes.deactivated_user: ["User has been deactivated"],
+        TestUserTypes.service_account: [
+            "Service accounts are unable to perform this action"
+        ],
+        TestUserTypes.site_admin: None,
+        TestUserTypes.superuser: None,
+    }
+
     should_succeed = expected_results[user_type]
 
-    print(response.content)
     if should_succeed:
         assert response.status_code == 200
         assert response.json()["categories"] == []
     else:
-        assert response.status_code == 403
-        assert (
-            response.json()["detail"]
-            == "You do not have permission to perform this action."
-        )
+        assert response.status_code == 400
+        assert response.json() == {"non_field_errors": error_message[user_type]}
```

Comment:
This should still continue to return a 403 error IMO. Also we don't want to leak information about _why_ the user doesn't have access to it, just that it doesn't exist. This can be considered a security problem (i.e. in the case of shadow bans or the like if we ever implement them, they could be used to infer information about something not considered public by us)